samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -70,6 +73,7 @@
   private SystemProducer systemProducer = null;
   private String key = null;
   private String streamName = null;
+  private int partitionCount = 0;
```

Comment:
Why is this a field on the StreamAppender?

Reply:
Lets the user set the partition count on the StreamAppender logging topic. In yarn, we could use the container count only, but in standalone deployments, we can't deduce the container count on startup. It was a suggestion from @jmakes and I think it's the only way to support both yarn and standalone deployments for now.